### ISSUE TITLE
"Forced unmount because devices don't match" every run for curlftpfs

### ISSUE THREAD

**Issue by uhriap** [saltstack/salt#53561]
### Description of Issue
mount.mounted state always force remount for curlftpfs mount with `Forced unmount because devices don't match.`
I believe this https://github.com/saltstack/salt/blob/8973ed2fd755d6579db6eee04a163c7db5735f36/salt/states/mount.py#L309 assumptions is false and `mount.active` actually returns string with curlftpfs#

Workaround `device_name_regex: ftp://xxx:xxx@example.com/` helps
```
[INFO    ] {u'mount': True, u'umount': u"Forced unmount because devices don't match. Wanted: curlftpfs#ftp://xxx:xxx@example.com/ (ftp://xxx:xxx@example.com/), current: curlftpfs#ftp://xxx:xxx@example.com/, /root/curlftpfs#ftp:/xxx:xxx@example.com"}
```
### Setup
```
# cat test.sls
curlftpfs:
  pkg:
    - installed

curlftpfs mount xxx:
  mount.mounted:
    - name: /opt/xxx
    - device: curlftpfs#ftp://xxx:xxx@example.com/
    - fstype: fuse
    - mount: True
    - mkmnt: True
    - persist: False
    - require:
      - pkg: curlftpfs
```

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
# salt-call --versions-report
Salt Version:
           Salt: 2018.3.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Dec  4 2017, 14:50:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.88-42
         system: Linux
        version: Ubuntu 16.04 xenial
```

**Comment by Arpithp** [comment]
----------
          ID: mount_dr_setup_filestore4
    Function: mount.mounted
        Name: /mnt/filestore_setup4
      Result: True
     Comment: Target was successfully mounted. Entry already exists in the fstab.
     Started: 09:31:31.547241
    Duration: 260.769 ms
     Changes:
              ----------
              mount:
                  True
              umount:
                  Forced unmount because devices don't match. Wanted: 10.190.130.178:/filestore4, current: 10.190.130.178://filestore4, /10.190.130.178:/filestore4

this below is my code
{% set availability_zone = grains['meta-data']['placement']['availability-zone'] %}
{% set devices = {
    'us-west-2a': {
        'filestore_setup': '10.190.130.178:/filestore',
        'filestore_setup1': '10.190.130.178:/filestore1',
        'filestore_setup2': '10.190.130.178:/filestore2',
        'filestore_setup3': '10.190.130.178:/filestore3',
        'filestore_setup4': '10.190.130.178:/filestore4'
    },
    'us-west-2b': {
        'filestore_setup': '10.190.190.205:/filestore',
        'filestore_setup1': '10.190.190.205:/filestore1',
        'filestore_setup2': '10.190.190.205:/filestore2',
        'filestore_setup3': '10.190.190.205:/filestore3',
        'filestore_setup4': '10.190.190.205:/filestore4'
    }
} %}

mount_dr_setup_filestore:
  mount.mounted:
    - name: /mnt/filestore_setup
    - device: '{{ devices[availability_zone].get("filestore_setup", "") }}'
    - fstype: nfs
    - opts: 'nfsvers=4.1,rsize=1048576,wsize=1048576,soft,timeo=600,retrans=2,noresvport'
    - persist: True
    - mkmnt: True

mount_dr_setup_filestore1:
  mount.mounted:
    - name: /mnt/filestore_setup1
    - device: '{{ devices[availability_zone].get("filestore_setup1", "") }}'
    - fstype: nfs
    - opts: 'nfsvers=4.1,rsize=1048576,wsize=1048576,soft,timeo=600,retrans=2,noresvport'
    - persist: True
    - mkmnt: True

mount_dr_setup_filestore2:
  mount.mounted:
    - name: /mnt/filestore_setup2
    - device: '{{ devices[availability_zone].get("filestore_setup2", "") }}'
    - fstype: nfs
    - opts: 'nfsvers=4.1,rsize=1048576,wsize=1048576,soft,timeo=600,retrans=2,noresvport'
    - persist: True
    - mkmnt: True

mount_dr_setup_filestore3:
  mount.mounted:
    - name: /mnt/filestore_setup3
    - device: '{{ devices[availability_zone].get("filestore_setup3", "") }}'
    - fstype: nfs
    - opts: 'nfsvers=4.1,rsize=1048576,wsize=1048576,soft,timeo=600,retrans=2,noresvport'
    - persist: True
    - mkmnt: True

mount_dr_setup_filestore4:
  mount.mounted:
    - name: /mnt/filestore_setup4
    - device: '{{ devices[availability_zone].get("filestore_setup4", "") }}'
    - fstype: nfs
    - opts: 'nfsvers=4.1,rsize=1048576,wsize=1048576,soft,timeo=600,retrans=2,noresvport'
    - persist: True
    - mkmnt: True


so basically even if the device matches I still get forced unmount, I know there is two slashes in the current one but I checked my fstab entry and its a single slash.

**Comment by fusionstream** [comment]
Upz. Having similar issues with this in OMV (which uses this) when drive letters move around after a reboot.
